### ISSUE TITLE
Is it possible to set different types of integrality constraints?

### ISSUE THREAD

**Issue by hlukashonak** [coin-or/python-mip#277]
Dear MIP Team,

thank you for the package! I am familiarizing myself with it and unfortunately I was not able to find the answer in the documentation.
Is it possible to set various integrality constraints modes in MIP? (like e.g. in [milp from scipy](https://scipy.github.io/devdocs/reference/generated/scipy.optimize.milp.html)):

> **integrality** 1D array_like, optional
Indicates the type of integrality constraint on each decision variable.
0 : Continuous variable; no integrality constraint.
1 : Integer variable; decision variable must be an integer within bounds.
2 : Semi-continuous variable; decision variable must be within bounds or take value 0.
3 : Semi-integer variable; decision variable must be an integer within bounds or take value 0.

I found only the _relax_ parameter of the _optimize_ function, but this is not enough for my problem (I actually want to implement the case 3 of the listed above)

Many thanks!

**Comment by rschwarz** [comment]
I don't think so. In the [documentation of `Var`](https://docs.python-mip.com/en/latest/classes.html#var), the property `var_type` just gives the options:
> (‘B’) BINARY, (‘C’) CONTINUOUS and (‘I’) INTEGER.

**Comment by sebheger** [comment]
It is not possible yet. The official gurobi python api supports the requested variable definition.

At this point, I have to remark, that the requested variable definition is just a "convenience" feature. You can use a pure integer variable and binary variable together with a set of constraints to model such semi-integer variables.

**Comment by mhechthz** [comment]
Is there any new information on this topic? CBC as well as SCIP support semi-continuous variables (all commercial solvers either) but I cannot model it using mip. Do I now need to move to Pyomo (which I desperately try to avoid) or will those types be added in near future?
